File: 题目集锦/python题目/BMI.cpp
#include <iostream>
#include <algorithm>
#include <cstdio>
#include <cstring>
using namespace std;

const int Max = 1000010;
int num, option;
double height, weight;
double boy_low_standard, boy_high_standard, girl_low_standard, girl_high_standard;

inline int read()
{
    int s = 0, f = 0;
    char ch = ' ';
    while(!isdigit(ch))
    {
        f |= (ch == '-');
        ch = getchar();
    }
    while(isdigit(ch))
    {
        s = (s << 3) + (s << 1) + (ch ^ 48);
        ch = getchar();
    }
    return (f) ? (-s) : (s);
}

class solution
{
    public:
        void boy(double bmi)
        {
            cout << "Boy:";
            if(bmi < boy_low_standard)
            {
                cout << "Too thin!" << endl;
            }
            else if(bmi >= boy_low_standard && bmi <= boy_high_standard)
            {
                cout << "Not too thin, or not too fat." << endl;
            }
            else
            {
                cout << "Too fat" << endl;
            }
        }

        void girl(double bmi)
        {
            cout << "Girl:";
            if(bmi < girl_low_standard)
            {
                cout << "Too thin!" << endl;
            }
            else if(bmi >= girl_low_standard && bmi <= girl_high_standard)
            {
                cout << "Not too thin, or not too fat." << endl;
            }
            else
            {
                cout << "Too fat" << endl;
            }
        }
};

int main()
{
    while(1)
    {
        num++;
        cout << "Case" << num << ':' << endl;
        cout << "You should input three variables next, you height(m), weight(kg), and if you are a boy or a girl" << endl;
        cin >> height >> weight;
        option = read();
        cout << "Because I cannot remember the standard, so I need you to input them. ', 'First in pust the lower limit, then the higher limit; First boys, then girls" << endl;
        cin >> boy_low_standard >> boy_high_standard >> girl_low_standard >> girl_high_standard;
        int bmi = weight * 1.0 / (height * height);
        solution object;
        if(option == 1) object.boy(bmi);
        if(option == 0) object.girl(bmi);
    }
    return 0;
}

File: 题目集锦/python题目/BMI.py
case = 0
while 1:
    case += 1
    print('Case', case, ':')
    print('You should input three variables next, you height(m), weight(kg), and if you are a boy or a girl')
    height = float(input())
    weight = float(input())
    option = int(input())

    print('Because I cannot remember the standard, so I need you to input them. ', 'First in pust the lower limit, then the higher limit; First boys, then girls')
    boy_low_standard = float(input())
    boy_high_standard = float(input())

    girl_low_standard = float(input())
    girl_high_standard = float(input())

    bmi = weight / (height * height)
    if(option == 1):
        print('Boy: ')
        if(bmi < boy_low_standard):
            print('Too thin!')
        elif(bmi >= boy_low_standard and bmi <= boy_high_standard):
            print('Not too thin, or not too fat.')
        else:
            print('Too fat')
    if(option == 0):
        print('Girl: ')
        if(bmi < girl_low_standard):
            print('Too thin!')
        elif(bmi >= girl_low_standard and bmi <= girl_high_standard):
            print('Not too thin, or not too fat.')
        else:
            print('Too fat')